Development: Settings calls

JohannPetzold opened the pull request:
- Add call for update email
- Add call for update password
- Add call for send email verification
- Add UserErrors
- Add tests with local database datas

JohannPetzold committed:
Adding Firebase calls for Settings

- Add call for update email
- Add call for update password
- Add UserErrors
- Add tests with local database datas

Sombre-Osmoze commented on Sources/SpotrLogic/SpotrLogic.swift:
```diff
@@ -230,6 +230,74 @@ public class SpotrLogic {
+    ///   - email: The user email.
+    ///   - password: The user password.
+    ///   - completion: The completion error result.
+    private func reauthenticate(with email: String, password: String, completion: @escaping (Error?) -> Void) {
+        let credential = EmailAuthProvider.credential(withEmail: email, password: password)
+        auth?.currentUser?.reauthenticate(with: credential, completion: { result, error in
+            completion(error)
+        })
+    }
```
Use the `handleAuthResult` (e.i: login)

Sombre-Osmoze reviewed (changes requested):
Resolve changes & test reauthentications failed tests

JohannPetzold committed:
Improving Settings calls

- Add support of handle result
- Add more tests to cover all cases
- Add accounts for testing
